src/ale.cpp

```diff
@@ -339,41 +174,36 @@ namespace ale {
      throw invalid_argument("At least two points must be input to interpolate over.");
    }
    if (points.size() != times.size()) {
-     throw invalid_argument("Invalid gsl_interp_type data, must have the same number of points as times.");
-   }
-   if (time < times.front() || time > times.back()) {
-     throw invalid_argument("Invalid gsl_interp_type time, outside of input times.");
+     throw invalid_argument("Must have the same number of points as times.");
    }
 
-   // convert our interp enum into a GSL one,
-   // should be easy to add non GSL interp methods here later
-   const gsl_interp_type *interp_methods[] = {gsl_interp_linear, gsl_interp_cspline};
-
-   gsl_interp *interpolator = gsl_interp_alloc(interp_methods[interp], numPoints);
-   gsl_interp_init(interpolator, &times[0], &points[0], numPoints);
-   gsl_interp_accel *acc = gsl_interp_accel_alloc();
+   int order;
+   switch(interp) {
+     case LINEAR:
+       order = 2;
+       break;
+     case SPLINE:
+       order = 4;
+       break;
+     case LAGRANGE:
+       order = 8;
+       break;
+     default:
+       throw invalid_argument("Invalid derivitive option, must be 0 or 1.");
```

Comment:
Should throw a different error message as this is selecting interp type not derivative.